pilot/cmd/pilot-agent/main.go

```diff
@@ -98,7 +99,7 @@ var (
 			// set values from registry platform
 			if len(role.IPAddress) == 0 {
 				if registry == serviceregistry.KubernetesRegistry {
-					role.IPAddress = os.Getenv("INSTANCE_IP")
+					role.IPAddress = pilot.ProxyIPAddress
```

Comment:
These can stay in current place. It’s not a feature flag. It’s just environment specific property for local Ip. Same goes for other env from the kubernetes downward Api

Reply:
make sense